src/safeposix/syscalls/fs_calls.rs

```diff
@@ -846,29 +934,87 @@ impl Cage {
         }
     }
 
-    //------------------------------------UNLINK SYSCALL------------------------------------
-
+    /// ## ------------------UNLINK SYSCALL------------------
+    /// ### Description
+    ///
+    /// The `unlink_syscall()` removes a link to a file. It removes the link
+    /// named by the pathname pointed to by path and decrements the link
+    /// count of the file referenced by the link.
+    /// If that name was the last link to a file and no processes have the file
+    /// open, the file is deleted and the space it was using is made
+    /// available for reuse. If the name was the last link to a file but any
+    /// processes still have the file open, the file will remain in
+    /// existence until the last file descriptor referring to it is closed.
+    /// On successful completion, the timestamp for the parent directory is
+    /// updated along with its linkcounts.
+    ///
+    /// ### Function Arguments
+    ///
+    /// The `unlink_syscall()` receives one argument:
+    /// * `path` - This argument points to a pathname which needs to be unlinked
+    ///
+    /// ### Returns
+    ///
+    /// Upon successful unlinking of the file, 0 is returned.
+    /// Otherwise, −1 is returned, and errno is set to indicate the error.
+    ///
+    /// ### Errors
+    ///
+    /// * `ENOENT` - The path argument is a null pathname;
+    /// a component of path prefix does not exist; or the file
+    /// named by oldpath does not exist.
+    /// * `EISDIR` - When the unlinking is done on a directory
+    ///
+    /// ### Panics
+    ///
+    /// * If the parent inode does not exist in the inode table, causing
+    ///   unwrap() to panic.
+    ///
+    /// For more detailed description of all the commands and return values, see
+    /// [unlink(2)](https://man7.org/linux/man-pages/man2/unlink.2.html)
     pub fn unlink_syscall(&self, path: &str) -> i32 {
+        // Return an error if the provided path is empty
         if path.len() == 0 {
```

Comment:
Out of curiosity, why special case this and fail fast?  You're going to figure this out in a few lines anyways...

Reply:
I think this might be to maintain consistency across other system calls, and its better to check for base cases first, before proceeding to the main implementation.